### PR TITLE
Feature/improve type info alias management

### PR DESCRIPTION
**Pull request by MagnaboscoL**
Implement a minor improvement to TypeInfo TypeId/names alias management:
- the same TypeInfo could be used for different compiler generated TypeId
- TypeInfoRepository::type now evaluate mtypenames TypeInfo too

### CODE REVIEW

File: rtt/ServiceRequester.cpp [hunk 1]
```diff
@@ -134,31 +134,34 @@ namespace RTT
...
     bool ServiceRequester::connectTo( Service::shared_ptr sp) {
-        for (OperationCallers::iterator it = mmethods.begin(); it != mmethods.end(); ++it) {
-            if ( !it->second->ready() ) {
-                if (sp->hasOperation( it->first )) {
-                    it->second->setImplementation( sp->getLocalOperation( it->first ), mrowner ? mrowner->engine() : 0 );
-                    if ( it->second->ready() ) {
-                        log(Debug) << "Successfully set up OperationCaller " << it->first <<endlog();
-                        if (!mrowner)
-                            log(Debug) << "OperationCaller "<< it->first << " has no caller set: using GlobalEngine."<<endlog();
+        if (sp)
```
Comment (meyerj) [inline rtt/ServiceRequester.cpp]: Better and without the need to indent the whole method:

``` cpp
bool ServiceRequester::connectTo( Service::shared_ptr sp) {
    if (!sp) return false;
    // same code as before
}
```

File: rtt/base/InputPortInterface.cpp [hunk 2]
```diff
@@ -84,12 +84,17 @@ InputPortInterface::NewDataOnPortEvent* InputPortInterface::getNewDataOnPortEven
 #endif
 bool InputPortInterface::connectTo(PortInterface* other, ConnPolicy const& policy)
 {
-    OutputPortInterface* output = dynamic_cast<OutputPortInterface*>(other);
-    if (! output) {
-        log(Error) << "InputPort "<< getName() <<" could not connect to "<< other->getName() << ": not an Output port." <<endlog();
-        return false;
+    if(other)
+    {
```
Comment (meyerj) [inline rtt/base/InputPortInterface.cpp:88]: Same here (with optional error logging).

File: rtt/types/TypeInfo.hpp [hunk 3]
```diff
@@ -478,9 +495,16 @@ namespace RTT
...
-        const char* mtid_name;
+        /** Main compiler generated TypeId that uses this TypeInfo */
         TypeId mtid;
+        /** Vector of the alternative compiler generated TypeId that can use
+         *  this TypeInfo
+         */
+        std::vector<TypeId> mtidalias;
+        /** The TypeId name generated by the compiler */
+        const char* mtid_name;
+        /** Vector of alias names that can be populated using addAlias() connectTo*/
```
Comment (meyerj) [inline rtt/types/TypeInfo.hpp]: connectTo?

File: rtt/ServiceRequester.cpp (hunk 1 above)
Comment (MagnaboscoL) [reply rtt/ServiceRequester.cpp]: Ok

File: rtt/base/InputPortInterface.cpp (hunk 2 above)
Comment (MagnaboscoL) [reply rtt/base/InputPortInterface.cpp:88]: Ok

File: rtt/types/TypeInfo.hpp (hunk 3 above)
Comment (MagnaboscoL) [reply rtt/types/TypeInfo.hpp]: Typo...fixed..

**Comment by MagnaboscoL** [comment]
A more efficient solution has been found

**Pull request closed without merging**